apps/local_laplacian/local_laplacian_generator.cpp

```diff
@@ -118,6 +158,30 @@ class LocalLaplacian : public Halide::Generator<LocalLaplacian> {
                 }
                 outGPyramid[j].compute_root().gpu_tile(x, y, xi, yi, blockw, blockh);
             }
+        } else if (get_target().features_any_of({Target::HVX_64, Target::HVX_128
+}) && !float_point) {
+            const int vector_size = get_target().has_feature(Target::HVX_128) ? 128 : 64;
+            int vec_lanes_16 = vector_size / 2;
+            int vec_lanes_8  = vector_size;
+            Var yo;
+            output.reorder(c, x, y).split(y, yo, y, 64).parallel(yo).vectorize(x, 8);
+            gray.compute_root().parallel(y, 32).vectorize(x, vec_lanes_16);
+            for (int j = 1; j < 5; j++) {
```

Comment:
It looks like this is running all of the big pyramid levels on the CPU (1-5), and all of the small pyramid levels (6+) on Hexagon. This is backwards from what I'd expect. The big pyramid levels are the expensive ones, the first 5 levels have literally >99.9% of the work (sum(1/4^k for 0 <= k < 5)/sum(1/4^k for k >= 0) = 0.99975...).

I would expect this schedule to run the first few levels on Hexagon, and then run the small levels on the CPU (where "small" is < a few HVX vectors). For typical image sizes (2-12 megapixels), that would mean maybe 2 or 3 levels running on Hexagon.